carl/ratios/classifier.py

```diff
@@ -149,15 +149,43 @@ def predict(self, X, log=False, **kwargs):
             The predicted ratio `r(X)`.
         """
         if self.identity_:
-            if log:
-                return np.zeros(len(X))
+            if not return_std:
+                if log:
+                    return np.zeros(len(X))
+                else:
+                    return np.ones(len(X))
             else:
-                return np.ones(len(X))
+                if log:
+                    return np.zeros(len(X)), np.zeros(len(X))
+                else:
+                    return np.ones(len(X)), np.zeros(len(X))
 
         else:
-            p = self.classifier_.predict_proba(X)
+            if not return_std:
+                p = self.classifier_.predict_proba(X)
+
+                if log:
+                    return np.log(p[:, 0]) - np.log(p[:, 1])
+                else:
+                    return np.divide(p[:, 0], p[:, 1])
 
-            if log:
-                return np.log(p[:, 0]) - np.log(p[:, 1])
             else:
-                return np.divide(p[:, 0], p[:, 1])
+                p, std_p = self.classifier_.predict_proba(X, return_std=True)
+
+                r = np.divide(p[:, 0], p[:, 1])
+                std_r = (r ** 2 * ((std_p / p[:, 0]) ** 2 +
+                                   (std_p / p[:, 1]) ** 2 -
+                                   2 * (-std_p ** 2) / (p[:, 0] *
+                                                        p[:, 1]))) ** 0.5
+                # nb: cov(p, 1-p) = -var(p) = -std^2(p)
```

Comment:
not sure if there are numerical issues.
pretty sure this simplifies to 
`std_r = std_p / ( p[:,1]**2)`